### PR TITLE
[website_event_excerpt_img] Better layout for website events.

### PR DESCRIPTION
**Pull request by yajo**
This displays an image taken from the event description and an excerpt in the events page, making it more attractive.

This one is quite similar to https://github.com/OCA/website/pull/163, but for events.

This is WIP because it will not work until these are merged:
- [x] https://github.com/OCA/server-tools/pull/355
- [x] https://github.com/OCA/server-tools/pull/354

The final result:

@rafaelbn @dgzurita

### CODE REVIEW

**Comment by yajo** [comment]
New commit allows to enable events occupying half or full row, like here:

captura de pantalla de 2016-04-01 18-08-17

Not perfect, but as close as I can get without fully overwriting the much improvable default design from core.

**Comment by yajo** [comment]
Since core Odoo's layout was so unworkable, I remade it. Less options, easier, beautier:

**Comment by rafaelbn** [comment]
cc @antespi

**Comment by pedrobaeza** [comment]
:+1:

**Comment by rafaelbn** [comment]
Functional tested
- Go to runbot (all modules)
- Create an Event
- Go to website yo Event's page
- 500: Internal Server Error
- The error could come when using toguether this module and website_event_filter_selector, in a base database this error occurs after installing website_event_filter_selector.

```
500: Internal Server Error
Error

Error message:

"'bool' object has no attribute '__getitem__'" while evaluating
"type['type'][0]"

QWeb
Traceback

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/website/models/ir_http.py", line 199, in _handle_exception
    response = super(ir_http, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_http.py", line 145, in _handle_exception
    return request._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/http.py", line 668, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/http.py", line 1372, in get_response
    result.flatten()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/http.py", line 1243, in flatten
    self.response.append(self.render())
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/http.py", line 1237, in render
    context=request.context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/website/models/ir_ui_view.py", line 173, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_ui_view.py", line 1022, in render
    return self.pool[engine].render(cr, uid, id_or_xml_id, qcontext, loader=loader, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 263, in render
    return self.render_node(self.get_template(id_or_xml_id, qwebcontext), qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 300, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 438, in render_tag_call
    d[0] = self.render_element(element, template_attributes, generated_attributes, d)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 300, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 447, in render_tag_call
    return self.render(cr, uid, template, d)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 263, in render
    return self.render_node(self.get_template(id_or_xml_id, qwebcontext), qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 300, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 438, in render_tag_call
    d[0] = self.render_element(element, template_attributes, generated_attributes, d)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 302, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 300, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 424, in render_tag_foreach
    ru.append(self.render_element(element, template_attributes, generated_attributes, copy_qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 323, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 285, in render_node
    for att, val in attrs:
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/website/models/ir_qweb.py", line 61, in render_att_att
    for att, val in super(QWeb, self).render_att_att(element, attribute_name, attribute_value, qwebcontext):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 359, in render_att_att
    return [(attribute_name[6:], self.eval(attribute_value, qwebcontext))]
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 207, in eval
    raise_qweb_exception(message="Could not evaluate expression %r" % expr, expression=expr, template=template)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 204, in eval
    return qwebcontext.safe_eval(expr)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/addons/base/ir/ir_qweb.py", line 89, in safe_eval
    return eval(expr, None, locals_dict, nocopy=True, locals_builtins=True)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3146717-41-7a8dbb/openerp/tools/safe_eval.py", line 314, in safe_eval
    return eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
QWebException: "'bool' object has no attribute '__getitem__'" while evaluating
"type['type'][0]"
```

**Comment by rafaelbn** [comment]
This module https://github.com/OCA/event/tree/8.0/event_multi_image add a field image_main to event.event



Are both modules compatibles? Must we have any caution in the README?

**Comment by rafaelbn** [comment]
I have found an error with images when using more than 1 translation in website.
- Go to runbot and install 2 more languages (Spanish and German for ex.)
- Set 1 language as default, for ex. Spanish
- Disable in setting -> website setting language "English"
- Create an Event and publish
- Go to website -> events and insert a block with image, save
- Check both translations Spanish and German, you will check both have same image
- Now go to Spanish and change image
- Check both transltation, check the error, the image are differents



This force the user to change the image as many times as languages per each event

**Comment by yajo** [comment]
Hi @rafaelbn.
- https://github.com/OCA/event/pull/41#issuecomment-218329477: Old bug, should be fixed now with the rebase.
- https://github.com/OCA/event/pull/41#issuecomment-218331890: There's https://github.com/OCA/event/pull/33/, but both modules are independent. A glue module could be made to avoid UX confusion when using both. The module we have in this PR extracts the image from the HTML body, not from any list.
- https://github.com/OCA/event/pull/41#issuecomment-218338913: That's Odoo core's bug. See https://github.com/odoo/odoo/issues/10095.

File: website_event_excerpt_img/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Excerpt + Image in Events",
+    "summary": "New layout for event summary, including an excerpt and image",
+    "version": "8.0.1.0.0",
+    "category": "Website",
+    "website": "http://www.antiun.com",
+    "author": "Antiun Ingeniería S.L., Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline website_event_excerpt_img/__openerp__.py]: Put here Tecnativa

**Comment by pedrobaeza** [comment]
cc @Tecnativa

**Comment by carlosdauden** [comment]
Attend https://github.com/OCA/event/pull/41#discussion_r77662368
:+1:

File: website_event_excerpt_img/views/event_event_view.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<openerp>
```
Comment (carlosdauden) [inline website_event_excerpt_img/views/event_event_view.xml]: <odoo>

File: website_event_excerpt_img/README.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Images
+------
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Rafael Blasco <rafabn@antiun.com>
```
Comment (rafaelbn) [inline website_event_excerpt_img/README.rst]: rafael.blasco@tecnativa.com

File: website_event_excerpt_img/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
```
Comment (rafaelbn) [inline website_event_excerpt_img/__init__.py]: Tecnativa

File: website_event_excerpt_img/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
```
Comment (rafaelbn) [inline website_event_excerpt_img/__openerp__.py]: Tecnativa

File: website_event_excerpt_img/__openerp__.py (hunk 1 above)
Comment (rafaelbn) [reply website_event_excerpt_img/__openerp__.py]: Tecnativa

File: website_event_excerpt_img/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
```
Comment (rafaelbn) [inline website_event_excerpt_img/models/__init__.py]: Tecnativa

File: website_event_excerpt_img/models/event.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
```
Comment (rafaelbn) [inline website_event_excerpt_img/models/event.py]: Tecnativa

**Comment by rafaelbn** [comment]
Thanks

**Pull request merged**